test/integration/integration.test.ts

```diff
@@ -453,12 +462,14 @@ describe('Integration tests', (): void => {
       })
 
       expect(outgoingPayment_.id).toBe(outgoingPayment.id)
+      expect(outgoingPayment_.receiveAmount.value).toBe(amountValueToSend)
+      expect(outgoingPayment_.sentAmount.value).toBe(amountValueToSend)
```

Comment:
This is in the final "Get Outgoing Payment step". @mkurapov I think it satisfies your suggestion here https://developers.mews.com/compile-time-functional-programming-in-typescript/ to verify the payment actually completes (already had this check in P2P as it happens).

I also tried checking for the incoming payment webhook but decided against it. I think this mostly checks the same thing and the MASE doesn't need to do anything with the incoming payment completed webhook. The incoming payment completed webhook is very slow (not really sure why that is). I had to poll for 40 seconds just for that webhook, which increases the test runtime by about 4x. If checking the outgoing payment is sufficient then I think we should just limit it to that.

Reply:
Also I'm glad you thought the article you linked was a suggestion from me because that is pretty cool 😄

Reply:
> Also I'm glad you thought the article you linked was a suggestion from me because that is pretty cool 😄

pffft. wrong link. it was a comment from the previous PR that I meant to link.

That was part of a rabbit hole of trying to figure out how one might define a ts type for a number range. Like:

```
someRgb: IntRange<0, 255> = 0 // OK
someRgb: IntRange<0, 255> = 256 // Error
```

TLDR is **just dont do that**, but it's kinda doable with some major caveats using this recursive method: https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range. Then I was curious about that article I accidentally shared because it could perhaps allow using arithmetic instead ofjust statically defining some upper/lower limit. TS as a full-fledged language is basically voodoo.

Reply:
> We could also in addition get the incoming payment, and confirm the receivedAmount there. What do you think?

Yeah that make sense. added.

Reply:
> TS as a full-fledged language is basically voodoo.

I do respect those people trying to do crazy types for fun, but yes, TS is just a DX tool 😆